experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -201,6 +209,52 @@ export class EditManager<
 		});
 	}
 
+	/**
+	 * Returns the sequence id of the oldest sequenced revertible commit on this branch.
+	 *
+	 * TODO: may be more performant to maintain the oldest revertible on the branches themselves
+	 * this should be tested and revisited once branches are supported
+	 */
+	private getOldestRevertibleSequenceId(
+		branch: SharedTreeBranch<TEditor, TChangeset>,
+	): SequenceId | undefined {
+		if (this.oldestRevertibleSequenceId !== undefined) {
+			return this.oldestRevertibleSequenceId;
+		}
+
+		let oldest: SequenceId | undefined;
+		for (const revision of branch.revertibleCommits()) {
+			if (oldest === undefined) {
+				oldest = this.trunkMetadata.get(revision)?.sequenceId;
+			} else {
+				const current = this.trunkMetadata.get(revision)?.sequenceId;
+				if (current !== undefined) {
+					oldest = minSequenceId(oldest, current);
+				}
+			}
+		}
+
```

Comment:
Try setting `this.oldestRevertibleSequenceId` in this function. The layout is:

```ts
if (this.oldestRevertibleSequenceId === undefined) {
	// Compute and set `this.oldestRevertibleSequenceId`
	// this.oldestRevertibleSequenceId = ...
	// It is okay if it is still undefined at the end
}

return this.oldestRevertibleSequenceId;
```

Then you don't need to set it on line 552.
Trunk eviction code does not check `this.oldestRevertibleSequenceId` directly - it gets it by calling this function `(this.getOldestRevertibleSequenceId())`.

Reply:
good call, I just changed this to an update function since we don't actually ever need to get and update at the same time